### ISSUE TITLE
High DPI scaling does not work in the Steam Snap

### ISSUE THREAD

**Issue by alexaschke** [canonical/steam-snap#74]
Hi team,
I was running on the latest Steam Snap 1.0.0.75 Edge and noticed that the "enlarge text option" in the Steam interface settings is not making a difference. 

**To Reproduce**
1. Install the latest Steam Edge Snap either from the CLI or Software Store on a computer with a High DPI display (2k or 4k).
2. Enable the enlarge text option in the Steam settings and restart.
3. Disable the enlarge text option in the Steam settings and restart. 
4. No increase in text font is noticeable between both options, it remains consistently small.

**Expected behavior**

If the High DPI scaling would work as expected, the fonts throughout the Steam app would be larger with the "enlarge text" setting enabled and smaller when disabled.

**Screenshots**

Screenshot of the Steam app with the "enlarge text" setting enabled on a 1440p display (quite small).
![Screenshot from 2022-10-27 18-09-46](https://user-images.githubusercontent.com/85647916/198355024-4f1a29fb-239c-4613-b103-84c608e0a829.png)

Screenshot of the Steam app with the "enlarge text" setting disabled on a 1440p display (no difference). 
![Screenshot from 2022-10-27 18-11-21](https://user-images.githubusercontent.com/85647916/198355263-c87b9bb2-ae9b-4070-b3d8-29b1c2212710.png)

**Environment**  
```
name:      steam
summary:   Launcher for the Steam software distribution service (Early Access)
publisher: Canonical✓
store-url: https://snapcraft.io/steam
contact:   https://github.com/canonical/steam-snap
license:   unset
description: |
  Steam is a software distribution service with an online store, automated
  installation, automatic updates, achievements, SteamCloud synchronized
  savegame and screenshot functionality, and many social features.
commands:
  - steam
snap-id:      NeoQngJVBf2wKC48bxnF2xqmfEFGdVnx
tracking:     latest/beta
refresh-date: yesterday at 16:00 BST
channels:
  latest/stable:    –                              
  latest/candidate: –                              
  latest/beta:      1.0.0.75 2022-10-07 (52) 365MB -
  latest/edge:      1.0.0.75 2022-10-19 (70) 378MB -
installed:          1.0.0.75            (52) 365MB -
```

```
snap    2.57.4+22.10ubuntu1
snapd   2.57.4+22.10ubuntu1
series  16
ubuntu  22.10
kernel  5.19.0-23-generic
```

Distro & version: Ubuntu 22.10 x86_64 
CPU:  Intel i9-9900K (16) @ 5.000GHz
GPU:  NVIDIA GeForce RTX 2080 Ti Rev. A
GPU Driver Version:  nvidia-driver-515 (proprietary, tested)

**Comment by ashuntu** [comment]
I'm unable to reproduce this on my 4k laptop, with both NVIDIA and Intel graphics.

**Comment by ashuntu** [comment]
In your steps to reproduce, you say you're on edge but your `snap info steam` log shows beta, have you tried both? It might also be worth trying snapd from edge, but I doubt that will be the source of the problem.

**Comment by alexaschke** [comment]
I have now switched to edge, see output of `snap info steam`:

name:      steam
summary:   Launcher for the Steam software distribution service (Early Access)
publisher: Canonical✓
store-url: https://snapcraft.io/steam
contact:   https://github.com/canonical/steam-snap
license:   unset
description: |
  Steam is a software distribution service with an online store, automated
  installation, automatic updates, achievements, SteamCloud synchronized
  savegame and screenshot functionality, and many social features.
commands:
  - steam.report
  - steam
snap-id:      NeoQngJVBf2wKC48bxnF2xqmfEFGdVnx
tracking:     latest/edge
refresh-date: today at 22:43 BST
channels:
  latest/stable:    –                              
  latest/candidate: –                              
  latest/beta:      1.0.0.75 2022-10-07 (52) 365MB -
  latest/edge:      1.0.0.75 2022-10-19 (70) 378MB -
installed:          1.0.0.75            (70) 378MB -

However when I run steam from the CLI, the following log record shows up a lot:

Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
Could not connect to X session manager: Could not open network socket
ExecCommandLine: "'/home/alex/snap/steam/common/.local/share/Steam/ubuntu12_32/steam' '-no-cef-sandbox'"
System startup time: 3.00 seconds
flock /usr/share/fonts/truetype/liberation2/LiberationSans-Regular.ttf LOCK_SH failed. errno = 13BuildCompleteAppOverviewChange: 183
RegisterForAppOverview 1: 10ms
RegisterForAppOverview 2: 10ms
flock /usr/share/fonts/truetype/liberation2/LiberationSans-Regular.ttf LOCK_SH failed. errno = 13Installing breakpad exception handler for appid(steam)/version(1666144119)
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed
Installing breakpad exception handler for appid(steam)/version(1666144119)
Warning: failed to set thread priority: set failed for -10: -1: setpriority() failed

**Comment by ZoopOTheGoop** [comment]
Okay I tested this (on the current `edge` along with the deb version) and it's a Steam issue, not a snap issue.

I discovered two facts, these apply regardless of whether you're using the deb or snap versions:
1. The DPI scaling doesn't seem to kick in at 1440p, this is a strange choice but it is indeed very small
2. It does kick in at 4K, but it only considers the primary monitor. I launched it on my 4K TV and it never did DPI scaling until I manually set the TV to my primary display and then restarted Steam.

So my diagnosis is status: unfortunate, but not something we can fix. I'd recommend taking it up with the [Steam for Linux repo](https://github.com/ValveSoftware/steam-for-linux)

**Comment by MartinVonReichenberg** [comment]
Confirming it here too:
Seems like Ubuntu and Arch Linux are not aware of **fractional** (150%) **scaling** on X-org **X11** GNOME session:
On SUSE it works _out-of-the-box_.

```
martin@Kingston-USB-Flash-Drive--Ubuntu-23-10 ~> inxi -F
System:
  Host: Kingston-USB-Flash-Drive--Ubuntu-23-10 Kernel: 6.5.0-9-generic
    arch: x86_64 bits: 64 Desktop: GNOME v: 45.0 Distro: Ubuntu 23.10 (Mantic
    Minotaur)
Machine:
  Type: Laptop System: LENOVO product: 82JQ v: Legion 5 Pro 16ACH6H
    serial: <superuser required>
  Mobo: LENOVO model: LNVNB161216 v: NO DPK serial: <superuser required>
    UEFI: LENOVO v: GKCN60WW date: 03/07/2023
Battery:
  ID-1: BAT0 charge: 83.7 Wh (100.0%) condition: 83.7/80.0 Wh (104.6%)
CPU:
  Info: 6-core model: AMD Ryzen 5 5600H with Radeon Graphics bits: 64
    type: MT MCP cache: L2: 3 MiB
  Speed (MHz): avg: 1881 min/max: 400/4280 cores: 1: 2716 2: 2633 3: 2229
    4: 400 5: 2205 6: 2516 7: 2128 8: 400 9: 400 10: 2396 11: 2530 12: 2020
Graphics:
  Device-1: NVIDIA GA106M [GeForce RTX 3060 Mobile / Max-Q] driver: nvidia
    v: 535.113.01
  Display: x11 server: X.Org v: 1.21.1.7 with: Xwayland v: 23.2.0 driver: X:
    loaded: nvidia unloaded: fbdev,modesetting,nouveau,vesa
    gpu: nvidia,nvidia-nvswitch resolution: 3408x2130
  API: OpenGL v: 4.6.0 NVIDIA 535.113.01 renderer: NVIDIA GeForce RTX 3060
    Laptop GPU/PCIe/SSE2
Audio:
  Device-1: NVIDIA GA106 High Definition Audio driver: snd_hda_intel
  Device-2: AMD ACP/ACP3X/ACP6x Audio Coprocessor driver: N/A
  Device-3: AMD Family 17h/19h HD Audio driver: snd_hda_intel
  API: ALSA v: k6.5.0-9-generic status: kernel-api
  Server-1: PipeWire v: 0.3.79 status: active
Network:
  Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet
    driver: r8169
  IF: eno1 state: up speed: 1000 Mbps duplex: full mac: 9c:2d:cd:c9:f6:9b
  Device-2: MEDIATEK MT7921 802.11ax PCI Express Wireless Network Adapter
    driver: mt7921e
  IF: wlp4s0 state: down mac: f0:a6:54:33:f4:2d
  Device-3: Realtek RTL8153 Gigabit Ethernet Adapter driver: r8152 type: USB
  IF: enx9405bb1970c2 state: down mac: 94:05:bb:19:70:c2
Bluetooth:
  Device-1: Foxconn / Hon Hai MediaTek Bluetooth Adapter driver: btusb
    type: USB
  Report: hciconfig ID: hci0 rfk-id: 2 state: down
    bt-service: enabled,running rfk-block: hardware: no software: yes
    address: F0:A6:54:33:F4:2E
Drives:
  Local Storage: total: 505.77 GiB used: 22.83 GiB (4.5%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: MZVLB512HBJQ-000L2
    size: 476.94 GiB
  ID-2: /dev/sda vendor: Kingston model: DataTraveler 3.0 size: 28.83 GiB
    type: USB
Partition:
  ID-1: / size: 27.17 GiB used: 22.82 GiB (84.0%) fs: ext4 dev: /dev/sda2
  ID-2: /boot/efi size: 1.05 GiB used: 6.1 MiB (0.6%) fs: vfat
    dev: /dev/sda1
Swap:
  ID-1: swap-1 type: file size: 4 GiB used: 0 KiB (0.0%) file: /swap.img
Sensors:
  System Temperatures: cpu: 42.2 C mobo: N/A gpu: nvidia temp: 39 C
  Fan Speeds (rpm): N/A
Info:
  Processes: 362 Uptime: 1h 0m Memory: total: 16 GiB available: 15.46 GiB
  used: 3.62 GiB (23.4%) Shell: fish inxi: 3.3.29
```

#### More shown in Pictures:
![Screenshot from 2023-10-24 16-24-26](https://github.com/canonical/steam-snap/assets/110736521/15ab5595-f4d6-4984-8d13-6a83434da358)
![Screenshot from 2023-10-24 16-32-09](https://github.com/canonical/steam-snap/assets/110736521/01fa0f84-a30f-406a-87e7-69eeb42a84a1)
![Screenshot from 2023-10-24 16-33-12](https://github.com/canonical/steam-snap/assets/110736521/c0cecbef-2d29-4610-91d6-50a907e11789)
![Screenshot from 2023-10-24 16-46-54](https://github.com/canonical/steam-snap/assets/110736521/64481076-62b3-4beb-b5e9-b5a6eda03eb3)

TL;DR:
**_There only two ways of enabling the Steam's fractional scaling along with the rest of the system:
Using `steam -forcedesktopscaling 1.5 %U` ARGUMENT
or
`env GDK_SCALE=2 steam` / `export GDK_SCALE=2 & steam`_**

**_They can be added to their respective application launchers in:
`/var/lib/snapd/desktop/applications/steam_steam.desktop` for the Snap version
`/usr/share/applications/steam.desktop` for the .DEBian version
`~/.local/share/applications/com.valvesoftware.Steam.desktop` for the Flatpak version_**
